packages/node/src/generators/setup-docker/setup-docker.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  addDependenciesToPackageJson,
+  convertNxGenerator,
+  formatFiles,
+  generateFiles,
+  GeneratorCallback,
+  joinPathFragments,
+  logger,
+  readNxJson,
+  readProjectConfiguration,
+  Tree,
+  updateProjectConfiguration,
+} from '@nrwl/devkit';
+import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
+import { SetUpDockerOptions } from './schema';
+
+function normalizeOptions(
+  tree: Tree,
+  setupOptions: SetUpDockerOptions
+): SetUpDockerOptions {
+  return {
+    ...setupOptions,
+    projectName: setupOptions.projectName ?? readNxJson(tree).defaultProject,
+    buildTarget: setupOptions.buildTarget ?? 'build',
+  };
+}
+
+function addDocker(tree: Tree, options: SetUpDockerOptions) {
+  const project = readProjectConfiguration(tree, options.projectName);
+  if (!project || !options.targetName) {
+    return;
+  }
+
+  if (tree.exists(joinPathFragments(project.root, 'DockerFile'))) {
+    logger.info(
+      `Skipping setup since a DockFile already exists inside ${project.root}`
+    );
+  }
+  const outputPath =
+    project.targets[`${options.buildTarget}`]?.options.outputPath;
+  generateFiles(tree, joinPathFragments(__dirname, './files'), project.root, {
+    tmpl: '',
+    app: project.sourceRoot,
+    buildLocation: outputPath,
+  });
+}
+
+export function updateProjectConfig(tree: Tree, options: SetUpDockerOptions) {
+  let projectConfig = readProjectConfiguration(tree, options.projectName);
+
+  const buildPort = Number(process.env.PORT) ?? 3000;
+
+  projectConfig.targets[`${options.targetName}`] = {
+    dependsOn: [`^${options.buildTarget}`],
+    executor: 'nx:run-commands',
+    options: {
+      commands: ['docker build .', `docker run -p ${buildPort}:${buildPort}`],
```

Comment:
We don't need to do `docker run` here since users can do that separately. Likely they will build and push the image to a registry rather than run it.

We could add another target like `docker-run` that can be configured as well (e.g. `--runTargetName`).